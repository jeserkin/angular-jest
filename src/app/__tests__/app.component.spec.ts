import { async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../app.component';
import { ConfigureFn, configureTests } from '@lib/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [AppComponent],
          imports: [RouterTestingModule, NoopAnimationsModule],
          schemas: [NO_ERRORS_SCHEMA]
        })
          .compileComponents();
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(AppComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('should create the app', async(() => {
    expect(component).toBeDefined()
  }));

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it(`should have as title 'frontend'`, async(() => {
    expect(component.title).toEqual('frontend 1');
  }));

  it('should render title in a h1 tag', async(() => {
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Welcome to frontend 1!');
  }));
});
