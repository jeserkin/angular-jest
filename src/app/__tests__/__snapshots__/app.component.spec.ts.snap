// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<my-app>
  <hello
    name=""
  />
  <p>
     Start editing to see some magic happen :)

  </p>
</my-app>
`;
